lib/internal/modules/esm/loader.js

```diff
@@ -372,15 +372,18 @@ class ModuleLoader {
 
     defaultLoadSync ??= require('internal/modules/esm/load').defaultLoadSync;
     const loadResult = defaultLoadSync(url, { format, importAttributes });
-    const {
-      format: finalFormat,
-      source,
-    } = loadResult;
+
+    let finalFormat = loadResult.format;
+    // Use the synchronous commonjs translator which can deal with cycles.
+    if (finalFormat === 'commonjs') {
+      finalFormat = 'commonjs-sync';
+    }
```

Comment:
```suggestion
    // Use the synchronous commonjs translator which can deal with cycles.
    let finalFormat = loadResult.format === 'commonjs' ? 'commonjs-sync' : loadResult.format;
```

Reply:
Personally I find ternary expressions harder to read than branches, and assignment with ternary expressions without parentheses is even worse. But I am okay with accepting the suggestions because I don't want to make the PR unnecessarily noisy.